truss/truss_handle.py

```diff
@@ -152,6 +152,60 @@ def get_docker_image(self, labels: Dict):
         """[Deprecated] Do not use."""
         return _docker_image_from_labels(labels)
 
+    @proxy_to_shadow_if_scattered
+    def run_python_script(
+        self,
+        script_path: Path,
+        build_dir: Optional[Path] = None,
+    ):
+        from python_on_whales.exceptions import DockerException
+
+        image = self.build_serving_docker_image(build_dir=build_dir)
+        secrets_mount_dir_path = _prepare_secrets_mount_dir()
+
+        envs: Dict[str, str] = {}
+        # Add bundled packages to the PYTHONPATH. Note
+        # that this is necessary to achieve the same environment as Truss Server
+        # -- this is setup that is done by Truss Server, that won't be available
+        # to the standalone script.
+        bundled_packages_path = Path("/packages")
+        envs["PYTHONPATH"] = bundled_packages_path.as_posix()
+
+        # Note that the entrypoint command should match
+        # what we use when executing Truss Server.
+        entrypoint_command = self.spec.python_executable_path or "python3"
+
+        def _docker_run(gpus: Optional[str] = None):
+            return Docker.client().run(
+                image.id,
+                entrypoint=entrypoint_command,
+                command=["/app/script.py"],
+                detach=False,
+                mounts=[
+                    [
+                        "type=bind",
+                        f"src={str(secrets_mount_dir_path)}",
+                        "target=/secrets",
+                    ],
+                    [
+                        "type=bind",
+                        f"src={str(script_path.absolute())}",
+                        "target=/app/script.py",
+                    ],
+                ],
+                gpus=gpus,
+                envs=envs,
+                add_hosts=[("host.docker.internal", "host-gateway")],
+                stream=True,
+            )
+
+        try:
+            return _docker_run(None)
+        except DockerException:
+            # The reason we'd wind up here is if the Truss needs
+            # a GPU, but the host does not have one that can attach.
+            return _docker_run(None)
```

Comment:
So should the `gpus` arg be not none in either of those cases? Right now try and except blocks are identical?

Reply:
oh yeah, in the first case it should be populated, let me try this again, thanks for catchign